Add techtree method in init, that return a dictionary with technology…

Systemac opened the pull request:
…, supply and facility necessary to obtain it.

Sorry for bad camelcase and variable'names...

Commit:
Add techtree method in init, that return a dictionary with technology, supply and facility necessary to obtain it.

Sorry for bad camelcase and variable'names...

Commit:
Add ressources.storage

PiecePaperCode commented on ogame/__init__.py:
```diff
@@ -319,6 +341,10 @@ class Resources:
             day_production = [int(day_production[0].span['title'].replace('.','')),
                               int(day_production[1].span['title'].replace('.','')),
                               int(day_production[2].span['title'].replace('.','')),]
+            storage = bs4.find_all('tr', attrs={'class':'alt'})[7].find_all('td', attrs={'class':'normalmark left2'})
```
i like that

PiecePaperCode commented on ogame/__init__.py:
```diff
@@ -298,6 +298,28 @@ def celestial_coordinates(self, id):
                 coordinates.append(const.destination.moon)
                 return coordinates
 
+    def techtree(self, id):
```
as mention in Discord we can save it offline and save the result in constnts.py

Systemac replied:
yes, but in another class, i don't want to break your code ^^

Commit:
Add price for research, a part of graviton,
Modify research of levels for bugfixes...
And add dictionnaries for techtree, the form is good, after I don't know how you want to implement it ...

Commit:
Bad login error, just fix it...

Commit:
Add techtrees method...

Systemac commented:
Done, if it's good for you...

Commit:
Bugfixes on storage, when resources is full.
Add method to know total, current and free fleet slot, idem for expedition.

Commit:
Bug fixes for news univers and attibute error.

Commit:
I removed techtree, I don't see how to implement it, I fixed some bugs and added the functionality of fleet movement numbers and available expedition numbers

Commit:
I removed techtree, I don't see how to implement it, I fixed some bugs and added the functionality of fleet movement numbers and available expedition numbers

Systemac commented:
End of bugfixes and abandon of treetech...

Commit:
Remove import of techtree.

Systemac commented:
End of bugfixes and abandon of treetech...

Commit:
maj

PiecePaperCode commented:
thx. for your work. im starting to i tegrate your techtree into the lib

Systemac commented:
> thx. for your work. im starting to i tegrate your techtree into the lib

Thx for you ^^

PiecePaperCode commented:
# Tests are all passing
```py
test_Constants (ogame.test.UnittestOgame) ... ok
test_Events (ogame.test.UnittestOgame) ... ok
test_Vars (ogame.test.UnittestOgame) ... ok
test_ally (ogame.test.UnittestOgame) ... ok
test_build (ogame.test.UnittestOgame) ... ok
test_celestial (ogame.test.UnittestOgame) ... ok
test_celestial_coordinates (ogame.test.UnittestOgame) ... ok
test_defences (ogame.test.UnittestOgame) ... ok
test_facilities (ogame.test.UnittestOgame) ... ok
test_fleet (ogame.test.UnittestOgame) ... ok
test_galaxy (ogame.test.UnittestOgame) ... ok
test_moon_facilities (ogame.test.UnittestOgame) ... ok
test_phalanx (ogame.test.UnittestOgame) ... ok
test_research (ogame.test.UnittestOgame) ... ok
test_resources (ogame.test.UnittestOgame) ... ok
test_send_fleet (ogame.test.UnittestOgame) ... ok
test_send_message (ogame.test.UnittestOgame) ... ok
test_ships (ogame.test.UnittestOgame) ... ok
test_spyreports (ogame.test.UnittestOgame) ... ok
test_supply (ogame.test.UnittestOgame) ... ok

----------------------------------------------------------------------
Ran 20 tests in 35.679s

OK

Process finished with exit code 0
```
